### DISCUSSION TITLE
search while using One-to-Many relation

### DISCUSSION THREAD

**Discussion by Hannestrand** [ldproxy/ldproxy#1311] [Configuration]
Hi

I have a type(call it A) that have a property(call it B) of the type OBJECT_ARRAY where i join two tables using One-to-Many relation. 
But any A where B contain more than one object will not get any hits while searching for fields in B. 

Is there anything special that is needed in the configuration for it to work? 
I am currently joining it this way
```       
propertyB:
        sourcePath: "[name=nameId]name_table"
        type: OBJECT_ARRAY
```

**Comment by zvfcs** [comment]
I'm also having trouble figuring out how to search objects inside an array. Since it shows up in the filter GUI, it gives the impression that it should be possible from there, but it's not.
![bilde](https://github.com/user-attachments/assets/56cb11f9-0075-413b-a21c-f104e16e3129)

I've also tried with cql2-text filter, but with no success. Of course, I'm not very fluent in cql2, so maybe that is the reason.

**Comment by azahnen** [comment]
I think you would have to use CQL2 array operations in this case. For example: `a_contains(propertyB.field,['value'])`

Also see https://docs.ogc.org/DRAFTS/21-065r1.html#_examples_3.

I will also check if we can support that in the filter GUI.

**Comment by cportele** [comment]
@azahnen - The operator for the filter query parameters - and, therefore, in the Filter GUI - is `A_EQUALS` for array-valued queryables.

**Reply by zvfcs** [reply]
I can't figure out how to do a query like this. 

https://api.ra.no/brukerminner/collections/brukerminner/items?filter=a_contains(bilder.lisens,['CC BY'])

gives me: _The parameter 'filter' is invalid: line 1:25 mismatched input '[' expecting {'(', Identifier}_

Reading the documentation cql2-documentation suggests that it should be "bilder:lisens", but the result is the same.

I get the same error with: 
https://api.ra.no/brukerminner/collections/brukerminner/items?filter=A_EQUALS(bilder:lisens,'CC BY') 

Is there anybody who can get it to work?

An example of the data I'm trying to query:

```
    "bilder": [
      {
        "url": "https://kms-api.kulturminnesok.no/image/xxl/files/uploads/1051126.rodesten-myggedalen-1.jpg",
        "fotograf": "Skjeberg Historielag v/Willy Tveter",
        "beskrivelse": "Rodestein, Myggedalen",
        "lisens": "CC BY",
        "opprettet": "2024-10-21T18:11:45Z",
        "endret": "2024-10-21T18:11:45Z"
      },
      {
        "url": "https://kms-api.kulturminnesok.no/image/xxl/files/uploads/1051126.rodesten-myggedalen-2.jpg",
        "fotograf": "Skjeberg Historielag v/Willy Tveter",
        "beskrivelse": "Rodestein, Myggedalen",
        "lisens": "CC BY",
        "opprettet": "2024-10-21T18:12:20Z",
        "endret": "2024-10-21T18:12:20Z"
      }
    ]
  }
```

**Reply by zvfcs** [reply]
Also, doesn't it make sense that this should work?: 
https://api.ra.no/brukerminner/collections/brukerminner/items?filter=bilder is not null

Gives: "The parameter 'filter' is invalid. Unknown or forbidden properties used: bilder."
